tools/cli/commands/changelog.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import chalk from 'chalk';
+import child_process from 'child_process';
+import path from 'path';
+
+/**
+ * Internal dependencies
+ */
+import promptForProject from '../helpers/promptForProject';
+import { chalkJetpackGreen } from '../helpers/styling';
+import { normalizeProject } from '../helpers/normalizeArgv';
+
+/**
+ * Command definition for the changelog subcommand.
+ *
+ * @param {object} yargs - The Yargs dependency.
+ *
+ * @returns {object} Yargs with the generate commands defined.
+ */
+export function changelogDefine( yargs ) {
+	yargs.command(
+		'changelog <cmd> [project]',
+		'Creates a new changelog file for project',
+		yarg => {
+			yarg
+				.positional( 'cmd', {
+					describe: 'Changelogger command (e.g. add)',
+					type: 'string',
+				} )
+				.positional( 'project', {
+					describe: 'Project in the form of type/name, e.g. plugins/jetpack',
+					type: 'string',
+				} );
+		},
+		async argv => {
+			await changeloggerCli( argv );
+		}
+	);
+
+	return yargs;
+}
+
+/**
+ * Runs changelogger script for project specified.
+ *
+ * @param {object} argv - arguments passed as cli.
+ */
+export async function changeloggerCli( argv ) {
+	// @todo Add validation of changelogger commands? See projects/packages/changelogger/README.md
+	// @todo refactor? .github/files/require-change-file-for-touched-projects.php to a common function that we could use here. Would allow us to run a "jetpack changelog add" without a project to walk us through all of them?
+	validateArgs( argv );
+	argv = normalizeProject( argv );
+	argv = await promptForProject( argv );
+	const projDir = path.resolve( `projects/${ argv.project }` );
+	const process = child_process.spawnSync( `vendor/bin/changelogger ${ argv.cmd }`, [ '' ], {
+		cwd: projDir,
+		stdio: 'inherit',
+		shell: true,
+	} );
+
+	// Node.js exit code status 0 === success
+	if ( process.status !== 0 ) {
+		console.error( chalk.red( 'Something went wrong! Check your file path?' ), process.error );
```

Comment:
You should also pass the exit code back out, so the caller of `jetpack changelog add` can know if the command failed or not.

I don't know if there's a better way than just calling `process.exit()` (after renaming your local `process` variable to not collide).

Reply:
Passed back out the status code if it fails. Not sure if that covers the same case as `process.exit()` but I think that plus the other error handling should hopefully cover more cases and provide more info. Open to more suggestions!